crypto/s2n_drbg.c

```diff
@@ -31,9 +31,9 @@
     acceptable in DRBG */
 int s2n_increment_drbg_counter(struct s2n_blob *counter)
 {
-    for (int i = counter->size - 1; i >= 0; i--) {
-        counter->data[i] += 1;
-        if (counter->data[i]) {
+    for (uint32_t i = counter->size; i > 0; i--) {
+        counter->data[i-1] += 1;
+        if (counter->data[i-1]) {
```

Comment:
@camshaft should we keep it like that or use `uint64_t`?

Reply:
`uint64_t` wouldn't help here since that would make the old loop guard (`i >= 0`) always evaluate to `true`.

However, we could use `int64_t` -- it would safely fit the worst case old initial value (`int64_t i = counter->size - 1` would not overflow even for `counter->size == UINT32_MAX`), and it would also allow for values below zero so the old loop guard `i >= 0` would be correct.

Reply:
another option is `ssize_t`, which is usually more correct for pointer offsets.

Reply:
I might be wrong, but isn't `ssize_t` only defined as part of POSIX, not the C standards? I see that it's being used in s2n quite a lot, but is it `#define`d by s2n somewhere? If not, (1) it may not compile on Windows, and (2) its width is platform-dependent and so a `UINT32_MAX` could still overflow on some platforms.

Reply:
This would be a non-issue if we used `ssize_t` (or something that's guaranteed to be narrower than `ssize_t`) for `counter->size` as well.

Reply:
> it may not compile on Windows

We have no support for windows so that's not really a concern. s2n's API is POSIX (or at least takes heavy inspiration from).

> its width is platform-dependent and so a UINT32_MAX could still overflow on some platforms.

If you're incrementing a pointer by a UINT32_MAX on a 32-bit platform it's going to overflow anyway.

Reply:
In the end, it's probably not that big of a deal. I think it's fine to go with what you have as it's better than what's there now.

Reply:
Thanks Cameron. I see! Then I agree `ssize_t` would be fine too.
I think I am also overthinking this and being too pedantic about the types!

I will go ahead and merge it. Thanks again @feliperodri and @camshaft.

Reply:
> I think I am also overthinking this and being too pedantic about the types!

I actually really appreciate these conversations; I think they're really important for building predictable software so no need to apologize :)